code/drasil-gool/lib/GOOL/Drasil/ClassInterface.hs

```diff
@@ -300,21 +300,31 @@ exists :: (ValueExpression r) => SValue r -> SValue r
 exists = notNull
 
 class (FunctionSym r) => InternalValueExp r where
+  -- | Generic function for calling an object method.
+  --   Takes the function name, the return type, the object, a list of 
+  --   positional arguments, and a list of named arguments.
   objMethodCallMixedArgs' :: Label -> VSType r -> SValue r -> [SValue r] -> 
     NamedArgs r -> SValue r
 
+-- | Default function for calling an object method.
+--   t is the return type of the method, o is the object, f is the method name,
+--   and ps is a list of positional arguments.
 objMethodCall :: (InternalValueExp r) => VSType r -> SValue r -> Label -> 
   [SValue r] -> SValue r
 objMethodCall t o f ps = objMethodCallMixedArgs' f t o ps []
 
+-- | Function for calling an object method with named arguments.
 objMethodCallNamedArgs :: (InternalValueExp r) => VSType r -> SValue r -> Label 
   -> NamedArgs r -> SValue r
 objMethodCallNamedArgs t o f = objMethodCallMixedArgs' f t o []
 
+-- | Function for calling an object method with a mix of positional
+--   and named arguments.
 objMethodCallMixedArgs :: (InternalValueExp r) => VSType r -> SValue r -> Label 
   -> [SValue r] -> NamedArgs r -> SValue r
 objMethodCallMixedArgs t o f = objMethodCallMixedArgs' f t o
 
+-- | Function for calling an object method with no parameters.
```

Comment:
I think the filename indicates that much of its content will be "utilities"-like already, so I think you can replace "Function for X" with just "X".

Is the "object" part of this needed? I think methods are always tied to objects. I _think_ Java's `public static` methods are more like functions than methods -- do we want to access them from these functions too?

Reply:
Ah, good point.  I think I noticed somewhere that there's a different function for calling static methods, so it is only for 'true' methods.